lib/eslint-patch.js

```diff
@@ -1,33 +1,50 @@
-'use strict';
-var meld = require('meld');
-var docs = require('./docs');
-var Config = require("eslint/lib/config");
-var ConfigUpgrader = require('./config_upgrader');
-
-var supportedPlugins = ['react', 'babel'];
-
-module.exports = function patcher(eslint) {
-
-  meld.around(eslint.CLIEngine, 'loadPlugins', function(joinPoint) {
-    var pluginNames = joinPoint.args[0];
-    var filteredPluginNames = pluginNames.filter(function(pluginName) {
-      return supportedPlugins.indexOf(pluginName) >= 0;
-    });
-    return joinPoint.proceed(filteredPluginNames);
-  });
-
-  meld.around(eslint.CLIEngine, 'addPlugin', function() {
-    return;
-  });
-
-  // meld.around(eslint.CLIEngine.Config, 'loadPackage', function(joinPoint) {
-  //   var filePath = joinPoint.args[0];
-  //   if (filePath.match(/^eslint-config-airbnb.*/)) {
-  //     return joinPoint.proceed();
-  //   } else {
-  //     return {};
-  //   }
-  // });
+"use strict";
+
+const Plugins = require("eslint/lib/config/plugins");
+const ModuleResolver = require("eslint/lib/util/module-resolver");
+
+const ConfigFile = require("eslint/lib/config/config-file");
+
+const Config = require("eslint/lib/config");
+const ConfigUpgrader = require("./config_upgrader");
+
+module.exports = function patch() {
+
+  const skippedModules = [];
+  function warnModuleNotSupported(name) {
+      if(skippedModules.indexOf(name) < 0) {
+        skippedModules.push(name);
+        console.error(`Module not supported: ${name}`);
+      }
+  }
+
+  const resolve = ModuleResolver.prototype.resolve;
+  ModuleResolver.prototype.resolve = function(name, path) {
+    try {
+      return resolve.apply(this, [name, path]);
+    } catch(e) {
+      warnModuleNotSupported(name);
+      return `${__dirname}/empty-plugin.js`;
+    }
+  };
+
+  Plugins.loadAll = function(pluginNames) {
+    const loadedPlugins = Object.keys(Plugins.getAll());
+    if(loadedPlugins.length > 0) {
+      return;
```

Comment:
This might be a wild guess, but it feels like this early return is the culprit of #214. If `loadAll` is called multiple times with different arguments then only the first invocation does something, while the others won't do anything (and maybe expected plugins won't be loaded/defined anymore), while the original implementation on ESLint does not checks for `getAll` and always attempt  to `load` each given plugin. 

Maybe we can monkey patch this again with only the `try`/`catch` around each `load` call and see how it goes. 🤔